src/fastga/models/performances/mission/dynamic_equilibrium.py

```diff
@@ -392,86 +403,56 @@ def equation_outer(
 
         return np.array([f1, f2])
 
+    def add_flight_point(self, flight_point: FlightPoint = None, equilibrium_result: tuple = None):
 
-def save_df(
-    time,
-    altitude,
-    distance,
-    mass,
-    v_tas,
-    v_cas,
-    rho,
-    gamma,
-    equilibrium_result,
-    thrust_rate,
-    sfc,
-    name: str,
-    existing_dataframe: pd.DataFrame = None,
-):
-    """
-    Method to save mission point to a pandas dataframe file for further post-processing
-
-    :param time: mission time in seconds
-    :param altitude: flight altitude in meters
-    :param distance: flight distance in meters
-    :param mass: aircraft current mass
-    :param v_tas: true air speed in m/s
-    :param v_cas: calibrated air speed in m/s
-    :param rho: air density in kg/m3
-    :param gamma: slope angle in degree
-    :param equilibrium_result: result vector of dynamic equilibrium
-    :param thrust_rate: thrust rate at flight point
-    :param sfc: sfc at flight point
-    :param name: phase name
-    :param existing_dataframe: Dataframe
-    """
+        """
+        Method to add single flight_point to a list of flight_point and treats equilibirum_result at the same time
+
+        :param equilibrium_result: result vector of dynamic equilibrium
+        :param flight_point: the flight_point to add
+        """
+
+        if flight_point is not None:
+            if equilibrium_result is not None:
+                flight_point.alpha = float(equilibrium_result[0]) * 180.0 / math.pi
+                flight_point.cl_wing = float(equilibrium_result[2])
+                flight_point.cl_htp = float(equilibrium_result[3])
+
+            self.flight_points.append(deepcopy(flight_point))
+
+    def complete_flight_point(
+        self, flight_point: FlightPoint, mach=None, v_cas=None, v_tas=None, climb_rate=0.0
+    ):
+
+        """
+        Method to complete velocity fields in flight_point. Uses ONE of [v_cas, v_tas, mach] velocity to set the others.
+        Order of priority is as presented in the list.
+
+        :param flight_point: the flight point to complete
+        :param mach: the mach number
+        :param v_cas: the calibrated airspeed
+        :param v_tas: the true airspeed
+        :param climb_rate: the climb rate in m/s used to compute gamma (can be negative).
+
+
+        """
+
+        atm = Atmosphere(flight_point.altitude, altitude_in_feet=False)
+
+        if v_cas is not None:
+            atm.calibrated_airspeed = v_cas
+
+        elif v_tas is not None:
+            atm.true_airspeed = v_tas
+
+        elif mach is not None:
+            atm.mach = mach
+
+        else:
+            raise ValueError("Either v_cas or v_tas must be given to complete flight_point")
```

Comment:
```suggestion
            raise ValueError("Either v_cas, v_tas or mach number must be given to complete flight_point")
```
Was the omission of the Mach number voluntary ?